### ISSUE TITLE
Pdflatex command was not found

### ISSUE THREAD

**Issue by horcsinbalint** [EotvosCollegium/mars#373]
When serving Mars's dynamic content from a Docker container running PHP-FPM, we had an issue with the `pdflatex` command not being found. The necessary packages were installed in the container and the command given to the `exec` function in `app/Console/Commands.php` were successful if I executed it inside the container in a `/bin/bash` terminal. Probably there is an issue that the PHP-FPM and bash has different `$PATH` values. As a quick fix, 5dc985ea3b16d1ca9d1cd8fe3fea895735432d18 contains `pdflatex`'s full path.

Most probably it is an issue with our deployment (giving PHP-FPM a wrong path), so there is probably not an issue with Mars. Potentially setting the commands in .env could fix the issue, but not setting a proper path could be attributed to user error.
So this issue is most probably invalid.

**Comment by kdmnk** [comment]
I found these issues (with #372, #366) slightly irrelevant as these should be a part of your next PR. Or do you want to have the deployment branch co-existing with development in the future? Because then we need to reconfigure other stuff as well.

For adding stuff into .env, just add it to .env.example and you can create a new config file for the server related stuff where you map them.

**Comment by horcsinbalint** [comment]
I added these issues so they would not be forgotten and I would like to include these changes in my upcoming PR, which would close these issues. In the past with other (mostly closed-source) projects we firstly created an issue for new features, and then developed it, and later created a PR, which closed these issues. (I might be mistaken, probably this is not how this project works.)

For the deployment and development branch: I would like them to co-exist. The deployment branch is the branch that is actually deployed on the server (in each deployment, that branch is cloned from the repo) and sometimes hotfixes are necessary that might be only compatible with our own environment (like #372). In this way, every change is version controlled, and it is not possible to make hot-fixes on the server itself. (So it takes time for changes to propagate. Which is both a bug and a feature: during this time somebody might realize a mistake.)
So I think new features and most changes should be done firstly in seperate branch, then after a PR, merged to the development branch, and then brought to the deployment branch. And the deployment branch can be used to make hotfixes that are not compatible with different deployments.

So I think the development branch should have a relatively clean history with properly managed PRs, and a deployment branch that might have some hotfixes that are necessary for the operation of our own deployment. But these divergencies should always be removed after some time.

It is probably a bad idea, and I think we should start a discussion about whether we should do this in a different manner. I don't know how these are usually done.
Best would be to always use properly written code and always deploy a stable branch but I think sometimes we do need temporary fixes.

I was considering creating a fork of the repo, but I think it would have just made our life harder.

**Comment by kdmnk** [comment]
> we firstly created an issue for new features, and then developed it, and later created a PR, which closed these issues. 

That should be the case here as well, I just don't see an issue stating the big picture, what are you currently working on in deployment. You can have sub-issues there. Referencing "random" commits in "random" branches is a bit confusing :D 

We had 2 different branches previously, separating deployment and development, but we did not take advantage of it and for convenience, now we only have one. I'd much like to see a proper workflow, similar to what you have described, if you use it for version control, or in the future for test servers, ci/cd, etc... I have not found it useful yet.

Forks are also not encouraged now, given that the bots don't have permission to push on your pr's then. But forks are just a matter of authorization policies.

**Comment by horcsinbalint** [comment]
I created issue #377 that contains a brief explanation about the new changes and I referenced the issues belonging to it, and added tags for them. I hope it clears up the issues a bit.

Thanks for the input, we might come back to having a single branch because it is easier to manage it that way (I had limited experience when I have split the branches). On the other hand, it would be nice if development and operational tasks were a bit segregated. Currently I am not working with development in the traditional sense (I am not making new features and I do not review pull requests), and I rather work with operational tasks (like fixing issues which affect day-to-day operations and I manage usually if new code needs to be deployed).
And some of my changes for the server might even break local debugging.
